calico_node/startup.go

```diff
@@ -14,59 +14,657 @@
+				message(" -  Updated IP: %s", ourIPv6)
+			}
+			continue
+		}
+
+		// Check that other nodes aren't using the same IP addresses.
+		// This is an error condition.
+		if theirIPv4 != nil && theirIPv4.Equal(ourIPv4.IP) {
+			message("Calico node '%s' is already using the IPv4 address %s:",
+				theirNode.Metadata.Name, ourIPv4)
+			message(" -  Check the node configuration to remove the IP address conflict")
+			errored = true
+		}
+		if theirIPv6 != nil && theirIPv6.Equal(ourIPv6.IP) {
+			message("Calico node '%s' is already using the IPv6 address %s:",
+				theirNode.Metadata.Name, ourIPv6)
+			message(" -  Check the node configuration to remove the IP address conflict")
+			errored = true
+		}
+	}
+
+	if errored {
+		terminate()
+	}
+}
+
+func ensureDefaultConfig(c *client.Client, node *api.Node) error {
+	// By default we set the global reporting interval to 0 - this is
+	// different from the defaults defined in Felix.
+	//
+	// Logging to file is disabled in the felix.cfg config file.  This
+	// should always be disabled for calico/node.  By default we log to
+	// screen - set the default logging value that we desire.
+	if err := ensureGlobalFelixConfig(c, "ReportingIntervalSecs", "0"); err != nil {
+		return err
+	} else if err = ensureGlobalFelixConfig(c, "LogSeverityScreen", client.GlobalDefaultLogLevel); err != nil {
+		return err
+	}
+
+	// Set the default values for some of the global BGP config values and
+	// per-node directory structure.
+	// These are required by both confd and the GoBGP daemon.  Some of this
+	// can only be done directly by the backend (since it requires access to
+	// datastore features not exposed in the main API).
+	//
+	// TODO: This is only required for the current BIRD and GoBGP integrations,
+	//       but should be removed once we switch over to a better watcher interface.
+	if err := ensureGlobalBGPConfig(c, "node_mesh", fmt.Sprintf("{\"enabled\": %v}", client.GlobalDefaultNodeToNodeMesh)); err != nil {
+		return err
+	} else if err := ensureGlobalBGPConfig(c, "as_num", strconv.Itoa(client.GlobalDefaultASNumber)); err != nil {
+		return err
+	} else if err = ensureGlobalBGPConfig(c, "loglevel", client.GlobalDefaultLogLevel); err != nil {
+		return err
+	} else if err = ensureGlobalBGPConfig(c, "LogSeverityScreen", client.GlobalDefaultLogLevel); err != nil {
+		return err
+	} else if err = c.Backend.EnsureCalicoNodeInitialized(node.Metadata.Name); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ensureGlobalFelixConfig ensures that the supplied global felix config value
+// is initialized, and if not initialize it with the supplied default.
+func ensureGlobalFelixConfig(c *client.Client, key, def string) error {
+	if val, assigned, err := c.Config().GetFelixConfig(key, ""); err != nil {
+		return err
+	} else if !assigned {
+		return c.Config().SetFelixConfig(key, "", def)
+	} else {
+		log.Infof("Global Felix value already assigned %s=%s", key, val)
```

Comment:
Can probably make this a debug log.

Also, do we have a way to configure the log level of the startup procedure itself?